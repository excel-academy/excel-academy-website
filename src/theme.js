import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
