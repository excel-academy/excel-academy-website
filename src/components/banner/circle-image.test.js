import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from 'rebass/styled-components';
import theme from '../../theme';
import CircleImageBanner from './circle-image';
import image from '../../utils/mockBannerImg';

describe('CircleImageBanner', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <CircleImageBanner image={image.src} imagedesc={image.alt} orient="right" p={{ sm: 1, tablet: 3 }}>
          <Heading fontWeight={300} fontSize="5rem" lineHeight={1} mb={1}>Ernie Bello</Heading>
        </CircleImageBanner>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
