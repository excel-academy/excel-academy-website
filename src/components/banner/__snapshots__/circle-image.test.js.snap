// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleImageBanner renders correctly 1`] = `
.c4 {
  box-sizing: border-box;
  margin: 0;
  margin-bottom: 2rem;
  font-size: 5rem;
  font-weight: 300;
  line-height: 1;
}

.c3 {
  box-sizing: border-box;
  z-index: 1;
}

.c6 {
  box-sizing: border-box;
  margin-bottom: 2rem;
  margin-left: 3rem;
  margin-right: 3rem;
  width: 15.75rem;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6 .gatsby-image-wrapper {
  height: 100%;
}

.c1 {
  box-sizing: border-box;
  padding: 2rem;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: hidden;
  text-align: left;
}

.c1 .c5 img {
  object-position: right !important;
}

.c0 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .c2 {
  max-width: 31.5rem;
}

.c0 .c5 {
  border-radius: 100%;
}

.c0 .c5 img {
  border-radius: 100%;
}

.c0 .c5 {
  margin-left: 0;
}

@media screen and (min-width:46em) {
  .c6 {
    width: 18.375rem;
  }
}

@media screen and (min-width:61.25em) {
  .c6 {
    width: 21rem;
  }
}

@media screen and (min-width:46em) {
  .c1 {
    padding: 4rem;
  }
}

@media screen and (orientation:portrait) {
  .c0 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    text-align: center;
  }

  .c0 .c5 {
    margin-right: 0;
    margin-left: 0;
  }
}

<section
  className="c0 c1"
  orient="right"
>
  <div
    className="c2 c3"
  >
    <h2
      className="c4"
      fontSize="5rem"
      fontWeight={300}
    >
      Ernie Bello
    </h2>
  </div>
  <div
    className="c5 c6"
    height={
      Object {
        "_": "15.75rem",
        "desktop": "21rem",
        "tablet": "18.375rem",
      }
    }
    width={
      Object {
        "_": "15.75rem",
        "desktop": "21rem",
        "tablet": "18.375rem",
      }
    }
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "66.66666666666667%",
            "width": "100%",
          }
        }
      />
      <picture>
        <img
          alt="banner"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 600px) 100vw, 600px"
          src="test-file-stub"
          srcSet="some srcSet"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" srcset=\\"some srcSet\\" src=\\"test-file-stub\\" alt=\\"banner\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
</section>
`;
